kura/org.eclipse.kura.core.deployment/src/main/java/org/eclipse/kura/core/deployment/hook/DeploymentHookManager.java

```diff
@@ -67,16 +68,17 @@ public synchronized void unbindHook(ServiceReference<DeploymentHook> hook) {
 
         updateAssociations();
         if (removedHook != null) {
-            context.getBundleContext().ungetService(hook);
+            getBundleContext().ungetService(hook);
             logger.info("Hook unregistered: {}", hookId);
         }
 
     }
 
-    public synchronized void activate(ComponentContext context) {
-        logger.info("activating...");
-        this.context = context;
-        logger.info("activating...done");
+    private BundleContext getBundleContext() {
```

Comment:
instead of having to do it through FrameworkUtil, you can add an activate method with BundleContext as argument. BundleContext instance will be injected by the runtime.

Reply:
The activate method was perviously there, with this PR I've removed it and switched to FrameworkUtil because DS can call bindHook() before activate(), this causes an NPE because the context field is null until the component is activated.

Reply:
Yeah you are right. I missed that. `bindHook()` is a service binding callback which is always called before activate. Instead of using `bindHook(ServiceReference<DeploymentHook> hook)` you can also use `bindHook(ServiceReference<DeploymentHook> hook, DeploymentHook deploymentHook)`. That would be beneficial instead of calling `context.getService()` inside `bind()`. In this case, you don't have to worry about resetting the service count by invoking `BundleContext#ungetService(ServiceReference)`

Reply:
Yes you are right, I'm using this signature for `bindHook()` becacuse I'm performing some validation on the `kura.service.pid` of the hook and I'm getting the service only if succeeds.

Switching to `bindHook(ServiceReference<DeploymentHook> hook, DeploymentHook deploymentHook)` would cause `getService()` to be called by DS even if the hook has an invalid `kura.service.pid`.

I'm performing this validation because `kura.service.pid` can be null for an hook if the property is not explicitly set in the component definition and `DeploymentHookManager` receives a reference to it before `ConfigurationService`, (this happened during my tests).

When you are done with the review can you please merge the PR? We are a bit in a hurry for the release.